Malmo/src/AgentHost.cpp

```diff
@@ -452,7 +453,8 @@ namespace malmo
     {
         boost::lock_guard<boost::mutex> scope_guard(this->world_state_mutex);
 
-        return this->world_state;
+        WorldState current_world_state(this->world_state);
```

Comment:
Does this do anything? This method is already returning a copy of `this->world_state`.

Reply:
It's to make clear that a copy is taken under the lock_guard which is then returned. My concern was that the compiler could read and return the value after the lock release and taking an explicit copy avoids the issue.

Reply:
Ah, good point! Never occurred to me. A comment might be useful, just to stop people like me "optimising" the code later on...

Reply:
Adding comment